local_planner/src/nodes/planner_functions.cpp

```diff
@@ -10,19 +10,26 @@ namespace avoidance {
 
 // trim the point cloud so that only points inside the bounding box are
 // considered
-void filterPointCloud(
-    pcl::PointCloud<pcl::PointXYZ>& cropped_cloud,
+void processPointcloud(
+    pcl::PointCloud<pcl::PointXYZI>& final_cloud,
     Eigen::Vector3f& closest_point, float& distance_to_closest_point,
     int& counter_backoff,
     const std::vector<pcl::PointCloud<pcl::PointXYZ>>& complete_cloud,
-    int min_cloud_size, float min_dist_backoff, Box histogram_box,
-    const Eigen::Vector3f& position, float min_realsense_dist) {
-  cropped_cloud.points.clear();
-  cropped_cloud.width = 0;
+    float min_dist_backoff, Box histogram_box, const Eigen::Vector3f& position,
+    float min_realsense_dist, int max_age) {
+  pcl::PointCloud<pcl::PointXYZI> old_cloud = final_cloud;
```

Comment:
Rather than copying and then resetting, simply do:
```
pcl::PointCloud<pcl::PointXYZI> old_cloud;
std::swap(final_cloud, old_cloud);
```

This avoids the copy of the points.